infra/test/Test/Pos/Infra/Arbitrary/Ssc.hs

```diff
@@ -0,0 +1,282 @@
+{-# LANGUAGE DeriveGeneric        #-}
+{-# LANGUAGE FlexibleContexts     #-}
+{-# LANGUAGE FlexibleInstances    #-}
+{-# LANGUAGE NoImplicitPrelude    #-}
+{-# LANGUAGE OverloadedStrings    #-}
+{-# LANGUAGE RecordWildCards      #-}
+{-# LANGUAGE ScopedTypeVariables  #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+-- | Arbitrary instances and generators for SSC types.
+
+module Test.Pos.Infra.Arbitrary.Ssc
+       ( SscPayloadDependsOnSlot (..)
+       , BadCommAndOpening (..)
+       , BadSignedCommitment (..)
+       , CommitmentOpening (..)
+
+       , commitmentMapEpochGen
+       , vssCertificateEpochGen
+       , genCommitmentsMap
+       , genSignedCommitment
+       , genVssCertificatesMap
+       , genSscPayload
+       , genSscPayloadForSlot
+       ) where
+
+import           Universum
+
+import qualified Data.List.NonEmpty as NE
+import           Test.QuickCheck (Arbitrary (..), Gen, choose, elements, listOf,
+                     oneof)
+import           Test.QuickCheck.Arbitrary.Generic (genericArbitrary,
+                     genericShrink)
+
+import           Pos.Core (EpochIndex, SlotId (..), VssCertificate (..),
+                     VssCertificatesMap, mkVssCertificate,
+                     mkVssCertificatesMapLossy)
+import           Pos.Core.Configuration (HasProtocolConstants,
+                     protocolConstants)
+import           Pos.Core.ProtocolConstants (ProtocolConstants (..),
+                     VssMaxTTL (..), VssMinTTL (..))
+import           Pos.Core.Ssc (Commitment (..), CommitmentsMap, Opening (..),
+                     SignedCommitment, SscPayload (..), SscProof (..),
+                     mkCommitmentsMap, randCommitmentAndOpening)
+import           Pos.Crypto (ProtocolMagic, SecretKey, deterministic,
+                     randomNumberInRange, toVssPublicKey, vssKeyGen)
+import           Pos.Infra.Communication.Types.Relay (DataMsg (..))
+import           Pos.Sinbin.Ssc.Base (isCommitmentIdExplicit,
+                     isOpeningIdExplicit, isSharesIdExplicit,
+                     mkSignedCommitment)
+import           Pos.Sinbin.Ssc.Message (MCCommitment (..), MCOpening (..),
+                     MCShares (..), MCVssCertificate (..), SscTag (..))
+import           Pos.Sinbin.Ssc.Toss.Types (TossModifier (..))
+import           Pos.Sinbin.Ssc.Types (SscGlobalState (..),
+                     SscSecretStorage (..))
```

Comment:
this import style is really growing on me :smile:

Reply:
You are welcome :)